### ISSUE TITLE
`./gradlew :checker:test` failing under JDK 23 due to new unnamed patterns test

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#7014]
See the [CI run](https://dev.azure.com/checkerframework/checkerframework/_build/results?buildId=17693&view=logs&j=3954a56b-cc4a-5fd5-c215-7e0918d09638&t=b646d790-9eed-5361-b917-f5bf7c9b693a&l=1811).  Also fails for me locally.

I think this is due to https://github.com/typetools/checker-framework/commit/2c7cc61851e699da75d10cf7345b085e2364297a.  The new `java22` tests don't cause crashes, but various checkers are reporting warnings on them; probably they need to be suppressed.

FYI @smillst

**Comment by mernst** [comment]
An example warning is:
```
UnnamedVars.java:47: error: [return] incompatible types in return.
    return winners;
           ^
  type of expression: @FenumUnqualified ArrayList<@FenumTop Object>
  method return type: @FenumUnqualified List<@FenumUnqualified Object>
1 error
```

**Comment by smillst** [comment]
I've pushed a fix for this to master.  @mernst Now that we have tests for Java 22+ language features, can you change it so that Java 23 is run in pull requests instead of Java 21?

**Comment by mernst** [comment]
@smillst Does that mean making a commit just like https://github.com/typetools/checker-framework/commit/118075e88a185107bbfe782adf27ee31068f43ef but that uses JDK23?

**Comment by msridhar** [comment]
Thanks!  I think this is fixed now so I will close.
